cli/connectible/remote.py

```diff
@@ -0,0 +1,146 @@
+            LOG.error(f"Failed to connect to host '{host}' with error -\n{str(e)}")
+            raise RemoteConnectionError(str(e))
+
+        LOG.info(f"Connected to host '{host}' successfully")
+        return client
+
+    def run(self, cmd, timeout=600):
+        """Execute command on host
+
+        Args:
+            cmd (str): Command to be executed
+            timeout (int): Socket timeout
+        """
+        stdout, stderr = "", ""
+        try:
+            # DEBUG.
+            LOG.info(f"[{self._host}] Executing command - {cmd}")
+
+            # Execuet command on node
+            _, stdout, stderr = self._client.exec_command(command=cmd, timeout=timeout)
+
+            # Format command output
+            stdout = "\n".join(map(lambda x: x.strip(), stdout.readlines()))
+            stderr = "\n".join(map(lambda x: x.strip(), stderr.readlines()))
+        except Exception as e:
+            LOG.error(f"Command '{cmd}' execution failed with error -\n{e}")
+            raise RemoteConnectionError(e)
+
+        return stdout, stderr
+
+    def run_async(self, cmd, interval=10, timeout=300):
+        """Execute command in background on host
+
+        Args:
+            cmd (str): Command to be executed
+            inverval (int): Interval to retry
+            timeout (int): Retry timeout
+        """
+        # Set return variables
+        stdout, stderr = "", ""
+
+        # DEBUG.
+        LOG.info(f"[{self._host}] Executing command in background - {cmd}")
+
+        try:
+            # Get transport channel and set configs
+            channel = self._client.get_transport().open_session()
+            channel.setblocking(0)
+
+            # Execute command on node
+            channel.exec_command(command=cmd)
+        except Exception as e:
+            LOG.error(f"Command '{cmd}' execution failed with error -\n{e}")
+            raise RemoteConnectionError(e)
+
+        # Wait for command execution and read output
+        for w in WaitUntil(timeout=timeout, interval=interval):
+            # Check for command status
+            if channel.exit_status_ready():
+                LOG.info(f"Command completed successfully within {timeout} sec")
+                break
+
+            # Check command output
+            while True:
+                try:
+                    out = channel.recv(1024)
+                    if not out:
+                        break
+                    stdout += out.decode()
+                except socket.timeout:
+                    break
+
+            LOG.info(f"Command still in progress, waiting for {interval} sec")
+
+        # Raise exception in case command still in progress
+        if w.expired:
+            raise UnexpectedStateError(
+                f"Failed to complete command within {timeout} sec"
+            )
+
+        # Read command error
+        while True:
+            try:
+                # Get 1kb of error
+                err = channel.recv_stderr(1024)
```

Comment:
One doubt here, do we need to add`.decode('ascii')`  for the data being received? or it is already in readable format?

Reply:
`recv_stderr` returns data in bytes. `err` is validated and converted to string at line #142